### PR TITLE
Added some features like username and password.

### PR DESCRIPTION
**Pull request by NicolasJourden**
Thanks.

### CODE REVIEW

File: mqttmpd.c [hunk 1]
```diff
@@ -584,6 +584,12 @@ int main(int argc, char *argv[])
 		break;
 	case 'p':
 		mpd_host = optarg;
+		str = strrchr(optarg, ':');
+		if (str > mpd_host && *(str-1) != ']') {
```
Comment (kurt-vd) [inline mqttmpd.c:603]: Why `*(str-1) != ']'`?
You hit the ipv6 closely, but I do not understand how your test works in such case (like `-p [::1]:6601` or so)

**Comment by kurt-vd** [comment]
I see your point, but I'd rather use a more complete uri parser that I already did in
https://github.com/kurt-vd/libet/blob/master/liburi.h

I'm also convinced that passwords should not **necessarily** be provided on the commandline.
That's why I created https://github.com/kurt-vd/libet/blob/master/libnetrc.h for another project.

I'll take a deeper look tonight, and see how I will merge those

File: mqttmpd.c (hunk 1 above)
Comment (kurt-vd) [reply mqttmpd.c:603]: stupid question of mine, you did the same as for MQTT :-)

**Comment by kurt-vd** [comment]
I looked at what you want to establish and created a branch `improve-connection-uris`

**Comment by kurt-vd** [comment]
Please evaluate this in your setup. I believe it should run fine.
It also gives you the possibility to put your password in `~/.netrc`, which is safer.

**Comment by kurt-vd** [comment]
I modified and cherry-picked your commit regarding the mosquitto username/password setup as it is mainly your contribution, I just stripped some parts which I believe are not necessary

**Comment by NicolasJourden** [comment]
Thanks for considering the pr.
I will have a look to new branch.

**Pull request closed without merging**